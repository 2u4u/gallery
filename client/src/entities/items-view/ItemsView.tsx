import { View } from 'shared/types/index.type';
import { FileType } from 'shared/model/reducers/appReducer';
import Grid from 'entities/grid';
import List from 'entities/list';

import './index.scss';

type Props = {
    data: FileType[];
    view: View;
    setModalOpen: (arg: boolean) => void;
    setModalContent: (arg: FileType) => void;
    setModalFileIndex: (arg: number) => void;
}
function ItemsView({ data, view, setModalOpen, setModalContent, setModalFileIndex }: Props) { 
    const handleOpenModal = (file: FileType, fileIndex: number) => {
        setModalOpen(true);
        setModalContent(file);
        setModalFileIndex(fileIndex);
    }
    
    return (
        <div className='items'>
            <div className='container'>
                {data.map((file) => {
                    const fileIndex = data.findIndex((dataFile) => dataFile._id === file._id);
                    return (
                        view === View.grid 
                            ? <Grid file={file} key={file._id} handleOpenModal={() => handleOpenModal(file, fileIndex)} /> 
                            : <List file={file} key={file._id} handleOpenModal={() => handleOpenModal(file, fileIndex)} />
                    )}
                )}
            </div>
        </div>
    );
}

export default ItemsView;
