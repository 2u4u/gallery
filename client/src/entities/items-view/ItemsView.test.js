import renderer from 'react-test-renderer';
import ItemsView from './ItemsView';
import { View } from 'shared/types/index.type';

describe('Items view component ', () => {
    it('matches snapshot', () => {
        const data = [
            {
                _id: 'someid',
                type: 'image',
                name: 'test image',
                url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
                duration: 'undefined',
                date: Date,
            }
        ]
        const view = View.grid;
        const setModalOpen = () => false;
        const setModalContent = () => false;
        const setModalFileIndex = () => false;
        const component = renderer.create(
            <ItemsView 
                data={data}
                view={view}
                setModalOpen={setModalOpen}
                setModalContent={setModalContent}
                setModalFileIndex={setModalFileIndex}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
