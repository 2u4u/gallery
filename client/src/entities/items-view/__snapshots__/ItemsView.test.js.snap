// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items view component  matches snapshot 1`] = `
<div
  className="items"
>
  <div
    className="container"
  >
    <div
      className="grid grid--image"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png)",
        }
      }
    >
      <div
        className="grid__content"
      />
    </div>
  </div>
</div>
`;
